src/libsync/discovery.cpp

```diff
@@ -719,7 +719,7 @@ void ProcessDirectoryJob::processFileAnalyzeLocalInfo(
 
     if (dbEntry.isValid()) {
         bool typeChange = localEntry.isDirectory != dbEntry.isDirectory();
-        if (!typeChange && localEntry.isVirtualFile) {
+        if (!typeChange && localEntry.isVirtualFile && dbEntry._modtime == localEntry.modtime && dbEntry._fileSize == localEntry.size) {
```

Comment:
since the "modtime and size haven't changed"-check is used twice I suggest to create an bool as `typeChange` above - might increase readability a bit.